packages/core/src/__tests__/state-manager.test.ts

```diff
@@ -71,75 +89,151 @@ describe('anchor', () => {
-    })
+    await ceramic.repository.stateManager.anchor(stream$)
     expect(stream$.value.log.length).toEqual(2)
   })
 
-  test('anchor retries successfully', async () => {
+  test(`_handleTip is retried until it returns`, async () => {
     const stateManager = ceramic.repository.stateManager
     const stream = await TileDocument.create(ceramic, INITIAL_CONTENT, null, { anchor: false })
     const stream$ = await ceramic.repository.load(stream.id, {})
 
+    await ceramic.repository.stateManager.anchor(stream$)
+    expect(stream$.value.anchorStatus).toEqual(AnchorStatus.PENDING)
+    await waitForAnchorStatus(stream$, AnchorStatus.ANCHORED)
+
     const fakeHandleTip = jest.fn()
-    ;(stateManager as any)._handleTip = fakeHandleTip
-    // Handle tip needs to work once to create the commit when the CAS calls applyCommit with it
-    fakeHandleTip.mockImplementationOnce(realHandleTip)
-    fakeHandleTip.mockRejectedValueOnce(new Error('Handle tip failed'))
+    const bound = fakeHandleTip.bind(stateManager)
+    ;(stateManager as any)._handleTip = bound
+
+    // Mock a throw as the first call
     fakeHandleTip.mockRejectedValueOnce(new Error('Handle tip failed'))
+
+    // Mock the result of the original implementation as the second call - this one should return
+    // and stop the retrying mechanism
     fakeHandleTip.mockImplementationOnce(realHandleTip)
 
-    await new Promise<void>((resolve) => {
-      ceramic.repository.stateManager.anchor(stream$).add(resolve)
-    })
-    expect(stream$.value.anchorStatus).toEqual(AnchorStatus.ANCHORED)
-    expect(fakeHandleTip).toHaveBeenCalledTimes(4)
+    // wait for 3 secs to let the retry mechanism work
+    await TestUtils.delay(3000)
+
+    // Check that fakeHandleTip was called only three times
+    expect(fakeHandleTip).toHaveBeenCalledTimes(2)
   })
 
-  test('anchor too many retries', async () => {
+  test(`_handleTip is retried up to three times, if it doesn't return`, async () => {
     const stateManager = ceramic.repository.stateManager
     const stream = await TileDocument.create(ceramic, INITIAL_CONTENT, null, { anchor: false })
     const stream$ = await ceramic.repository.load(stream.id, {})
 
+    await ceramic.repository.stateManager.anchor(stream$)
+    expect(stream$.value.anchorStatus).toEqual(AnchorStatus.PENDING)
+    await waitForAnchorStatus(stream$, AnchorStatus.ANCHORED)
+
     const fakeHandleTip = jest.fn()
+    fakeHandleTip.bind(stateManager)
     ;(stateManager as any)._handleTip = fakeHandleTip
+
+    // Mock a bunch of throws as consecutive calls to fakeHandleTip
     fakeHandleTip.mockRejectedValueOnce(new Error('Handle tip failed'))
     fakeHandleTip.mockRejectedValueOnce(new Error('Handle tip failed'))
     fakeHandleTip.mockRejectedValueOnce(new Error('Handle tip failed'))
-    fakeHandleTip.mockImplementationOnce(realHandleTip)
+    fakeHandleTip.mockRejectedValueOnce(new Error('Handle tip failed'))
```

Comment:
is there significance to there being 6 mock calls here?  Or is it just supposed to be "more than 3 failures since we expect 3 retries"?  If the latter, we could just use `fakeHandleTip.mockRejectedValue` which will return the given failed value no matter how many times it is called